packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -33,26 +34,69 @@ function getNodeProperty(node, path) {
 }
 
 /**
- * Get a normalized file path. If Embroider prefix is present, strip it out
+ * Function to parse defaults.project and only return info to be used by the plugin
+ * @param {object} project Ember defaults.project
+ * @returns {object} Contains project name and ember-addon path info
+ */
+function getProjectConfiguration(project) {
+  const parsedProjectConfiguration = {};
+
+  if (project) {
+    parsedProjectConfiguration.pkg = {
+      name: project.pkg.name,
+      'ember-addon': {
+        paths: project.pkg['ember-addon'].paths,
+      },
+    };
+  }
+
+  return parsedProjectConfiguration;
+}
+
+function _getParsedClassicFilepath(pathSegments, projectConfiguration) {
+  const projectNamePathSeparators = projectConfiguration.pkg.name.split(path.sep);
+
+  pathSegments.splice(
+    0,
+    pathSegments.indexOf(projectNamePathSeparators[0]) + projectNamePathSeparators.length
+  );
+
+  return pathSegments.join(path.sep);
+}
+
+function _getParsedEmbroiderFilepath(pathSegments) {
+  const RELATIVE_PATH_ROOT = 2;
+
+  // Strip out Embroider prefix (embroider/nnnnnn) segments from file path
```

Comment:
If we're going to put a comment here, let's make it more complete. Let's indicate what a typical `pathSegments` array would look like to better indicate what we're doing to it.